umap/parametric_umap.py

```diff
@@ -1019,3 +1044,72 @@ def _parametric_reconstruction_loss(self, y, y_pred):
         loss = self.parametric_reconstruction_loss_fn(
             y["reconstruction"], y_pred["reconstruction"])
         return loss * self.parametric_reconstruction_loss_weight
+
+##################################################
+# 1. Pytorch version of parametric UMAP network. #
+##################################################
+
+
+class PumapNet(nn.Module):
```

Comment:
I think you need to wrap this class definition in a conditional on torch.nn having been imported -- it is currently causing the tests to fail in the CI.